### ISSUE TITLE
Header Group broken when Row Expand Indicator is enabled

### ISSUE THREAD

**Issue by PeterMerkert** [AllenFang/react-bootstrap-table#1417]
Hi,

when using the Row Expand Indicator option in conjunction with the header grouping, the header columns end up being wrongly set up. Have a look at the minimal example broken code here with the typical products.

```
class BrokenTable extends React.Component {
  render() {
    return (
      <BootstrapTable data={products}
                            expandableRow={(r) => {return true;}}
                            expandComponent={(r) => {return (<span>hello world</span>);}}
                            expandColumnOptions={ { expandColumnVisible: true } }>
        <TableHeaderColumn row='0' rowSpan='2' dataField='id' isKey>ID</TableHeaderColumn>
        <TableHeaderColumn row='0' colSpan='2'>Product</TableHeaderColumn>
        <TableHeaderColumn row='1' dataField='name'>Name</TableHeaderColumn>
        <TableHeaderColumn row='1' dataField='price'>Price</TableHeaderColumn>
      </BootstrapTable>
    );
  }
}

const products = [];

function addProducts(quantity) {
  const startId = products.length;
  for (let i = 0; i < quantity; i++) {
    const id = startId + i;
    if (i < 3) {
      products.push({
        id: id,
        name: 'Item name ' + id,
        price: 2100 + i,
        expand: [ {
          fieldA: 'test1',
          fieldB: (i + 1) * 99,
          fieldC: (i + 1) * Math.random() * 100,
          fieldD: '123eedd' + i
        }, {
          fieldA: 'test2',
          fieldB: i * 99,
          fieldC: i * Math.random() * 100,
          fieldD: '123eedd' + i
        } ]
      });
    } else {
      products.push({
        id: id,
        name: 'Item name ' + id,
        price: 2100 + i
      });
    }
  }
}

addProducts(5);
```

And the resulting table looks as follows:
![selection_015](https://user-images.githubusercontent.com/4216454/27472300-1a307518-57fb-11e7-8086-a58926412dcf.png)

**Comment by vincejairam** [comment]
I'm actually having the same issue and was just about to post, is there a possible workaround?

**Comment by AllenFang** [comment]
I'll fix it in next version, thanks @PeterMerkert your good example to reproduce 👍

**Comment by AllenFang** [comment]
Fixed on `v3.4.2`, please have a look on it and let me know if the problem still remain. thanks

**Comment by PeterMerkert** [comment]
Just update my project to `3.4.2`. Works smoothly! :+1: 

Massive thanks for the quick fix @AllenFang !
